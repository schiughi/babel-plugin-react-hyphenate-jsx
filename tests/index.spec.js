import { transform } from '@babel/core';
import fs from "fs";
import plugin from '../src';

const FIXTURES = "./tests/__fixtures__";

// specs
describe('babel-plugin-react-hyphenate-jsx', () => {
  it('should replace hyphen case args to camelCase in arrow function', () => {
    const example = fs.readFileSync(`${FIXTURES}/arrow-function.jsx`, 'utf8');
    const { code } = transform(example, {
      presets: ["@babel/preset-env","@babel/preset-react"],
      plugins: [plugin]
    });
    expect(code).toMatchSnapshot();
  });
  it('should replace hyphen case args to camelCase in class', () => {
    const example = fs.readFileSync(`${FIXTURES}/class-component.jsx`, 'utf8');
    const { code } = transform(example, {
      presets: ["@babel/preset-env","@babel/preset-react"],
      plugins: [plugin]
    });
    expect(code).toMatchSnapshot();
  });
});
