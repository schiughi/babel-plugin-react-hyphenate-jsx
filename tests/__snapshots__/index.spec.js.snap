// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`babel-plugin-react-hyphenate-jsx should replace hyphen case args to camelCase in arrow function 1`] = `
"\\"use strict\\";

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});
exports.Button = void 0;

var Button = function Button(_ref) {
  var onSubmit = _ref.onSubmit;
  return React.createElement(\\"button\\", {
    onClick: onSubmit
  }, \\"submit\\");
};

exports.Button = Button;"
`;

exports[`babel-plugin-react-hyphenate-jsx should replace hyphen case args to camelCase in class 1`] = `
"\\"use strict\\";

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require(\\"react\\"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _typeof(obj) { if (typeof Symbol === \\"function\\" && typeof Symbol.iterator === \\"symbol\\") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === \\"function\\" && obj.constructor === Symbol && obj !== Symbol.prototype ? \\"symbol\\" : typeof obj; }; } return _typeof(obj); }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError(\\"Cannot call a class as a function\\"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if (\\"value\\" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

function _possibleConstructorReturn(self, call) { if (call && (_typeof(call) === \\"object\\" || typeof call === \\"function\\")) { return call; } return _assertThisInitialized(self); }

function _getPrototypeOf(o) { _getPrototypeOf = Object.setPrototypeOf ? Object.getPrototypeOf : function _getPrototypeOf(o) { return o.__proto__ || Object.getPrototypeOf(o); }; return _getPrototypeOf(o); }

function _inherits(subClass, superClass) { if (typeof superClass !== \\"function\\" && superClass !== null) { throw new TypeError(\\"Super expression must either be null or a function\\"); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, writable: true, configurable: true } }); if (superClass) _setPrototypeOf(subClass, superClass); }

function _setPrototypeOf(o, p) { _setPrototypeOf = Object.setPrototypeOf || function _setPrototypeOf(o, p) { o.__proto__ = p; return o; }; return _setPrototypeOf(o, p); }

function _assertThisInitialized(self) { if (self === void 0) { throw new ReferenceError(\\"this hasn't been initialised - super() hasn't been called\\"); } return self; }

var Form =
/*#__PURE__*/
function (_React$Component) {
  _inherits(Form, _React$Component);

  function Form(props) {
    var _this;

    _classCallCheck(this, Form);

    _this = _possibleConstructorReturn(this, _getPrototypeOf(Form).call(this, props));
    _this.state = {
      email: \\"\\",
      password: \\"\\"
    };
    _this.handleChange = _this.handleChange.bind(_assertThisInitialized(_assertThisInitialized(_this)));
    _this.handleSubmit = _this.handleSubmit.bind(_assertThisInitialized(_assertThisInitialized(_this)));
    return _this;
  }

  _createClass(Form, [{
    key: \\"handleChange\\",
    value: function handleChange(event) {
      this.setState(_defineProperty({}, event.target.name, event.target.value));
    }
  }, {
    key: \\"handleSubmit\\",
    value: function handleSubmit() {
      this.props.onSubmit(this.state);
    }
  }, {
    key: \\"render\\",
    value: function render() {
      _react.default.createElement(\\"form\\", {
        acceptCharset: \\"UTF-8\\",
        onSubmit: this.handleSubmit
      }, _react.default.createElement(\\"input\\", {
        type: \\"email\\",
        name: \\"email\\",
        className: \\"form__input-email\\",
        value: this.state.email,
        onChange: this.handleChange
      }), _react.default.createElement(\\"input\\", {
        type: \\"password\\",
        name: \\"password\\",
        className: \\"form__input-password\\",
        value: this.state.password,
        onChange: this.handleChange
      }), _react.default.createElement(\\"button\\", {
        type: \\"submit\\"
      }, \\"submit\\"));
    }
  }]);

  return Form;
}(_react.default.Component);

exports.default = Form;"
`;
